lib/Mojo/mysql/Native/Database.pm

```diff
@@ -0,0 +1,227 @@
+package Mojo::mysql::Native::Database;
+use Mojo::Base 'Mojo::mysql::Database';
+
+use Mojo::mysql::Connection;
+use Mojo::mysql::Native::Results;
+use Mojo::mysql::Native::Transaction;
+use Mojo::mysql::Util qw(expand_sql parse_url);
+use Scalar::Util 'weaken';
+use Carp 'croak';
+
+has 'connection';
+
+sub DESTROY {
+  my $self = shift;
+  return unless my $c = $self->connection;
+  return unless my $mysql = $self->mysql;
+  $mysql->_enqueue($c);
+}
+
+sub backlog { scalar @{shift->{waiting} || []} }
+
+sub begin {
+  my $self = shift;
+  $self->query('START TRANSACTION');
+  $self->query('SET autocommit=0');
+  my $tx = Mojo::mysql::Native::Transaction->new(db => $self);
+  weaken $tx->{db};
+  return $tx;
+}
+
+sub connect {
```

Comment:
Using **url** everywhere where we need connect info is not a bad idea.
If we do this we have to either deprecate **username** and **password** in mysql.pm
or update the url when username or password is set.